services/web-app/components/medium-posts/medium-posts.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { Column, Grid } from '@carbon/react'
+import moment from 'moment'
+import PropTypes from 'prop-types'
+import { useEffect, useRef, useState } from 'react'
+
+import ArticleCard from '@/components/article-card'
+import Image from '@/components/markdown/image'
+import { mediaQueries, useMatchMedia } from '@/utils/use-match-media'
+
+import styles from './medium-posts.module.scss'
+
+const dummyArticles = [
+  {
+    title: 'Dummy Title',
+    author: 'Jane Doe',
+    link: '',
+    date: moment(new Date()).format('MMMM Do, YYYY'),
+    thumbnail: '/carbon.png'
+  },
+  {
+    title: 'Dummy Title',
+    author: 'Jane Doe',
+    link: '',
+    date: moment(new Date()).format('MMMM Do, YYYY'),
+    thumbnail: '/carbon.png'
+  },
+  {
+    title: 'Dummy Title',
+    author: 'Jane Doe',
+    link: '',
+    date: moment(new Date()).format('MMMM Do, YYYY'),
+    thumbnail: '/carbon.png'
+  }
+]
+
+const imgSrcMatchString = /<img[^>]+src="?([^"\s]+)"?\s*\/>/g
+
+const MediumPosts = ({ postLimit = 3, cardProps, dummy, ...rest }) => {
+  const isCancelled = useRef(false)
+  const [allPosts, setAllPosts] = useState(dummy ? dummyArticles : [])
+
+  const isMd = useMatchMedia(mediaQueries.md)
+
+  useEffect(() => {
+    async function getMediumArticles() {
+      const articlesRes = await fetch('http://localhost:3000/api/get-medium-feed') // TODO: replace this with graphql endpoint call
```

Comment:
Once we have this on the data graph, we should call it in gSSP or gSP with a revalidation interval :)

Reply:
this is a component , not a page so it doesn't take gssp 😕

Reply:
Oh haha. Good point. Then I guess passing it in as a prop from the containing page?

Reply:
yeah, depends if we'll consume it from a local mdx file or remote. if we do local I think we can add a gssp to it and consume the data then pass as prop

Reply:
Thinking about this some more based on the current state of our repo and mdx stuff...

In a similar way to how we'd talked about discovering images in MDX, optimizing them, and storing them as source sets "on the side" for later, this seems like a similar type of use case: MDX is ingested, certain nodes in the AST require additional processing before display, processing happens, AST nodes are updated accordingly based on the results, updated MDX (or JSX) is passed on through to the web-app from which it was originally requested.

This might put us back into the realm of needing an actual MDX service... Something that could be asked for "ready-to-go" MDX, and behind the scenes could be parsing it, calling out to other services to resolve/generate data, and compiling the results before sending back to the web-app.

Relative to this PR, discovering a `MediumPosts` component during MDX processing could trigger a query on the data-graph to get the post data (i.e. the props needed to render the list of medium articles).

Then we'd benefit from any caching done by the medium service to speed things up. 🚀 

The overall benefit of doing it like that as opposed to via fetch request in an effect is that it could be server-side rendered, which would be good for cross-linking and SEO. (and not needing skeletons while the content is loading).

Reply:
If it were up to me i'd go the medium-service -> gssp route just for the sake of simplicity, is there other advantages to doing the mdx-processing service instead?